File: 121买卖股票的最佳时机/max_profit.cpp
#include <algorithm>
#include <climits>
#include <iostream>
#include <vector>

int maxProfit(const std::vector<int>& prices) {
    int min_price = INT_MAX;
    int max_profit = 0;
    for (int p : prices) {
        if (p < min_price) {
            min_price = p;
        } else {
            max_profit = std::max(max_profit, p - min_price);
        }
    }
    return max_profit;
}

int main() {
    std::ios::sync_with_stdio(false);
    std::cin.tie(nullptr);

    std::vector<int> prices;
    int x;
    while (std::cin >> x) {
        prices.push_back(x);
    }
    std::cout << maxProfit(prices) << '\n';
    return 0;
}

File: 121买卖股票的最佳时机/max_profit.py
from typing import List


def max_profit(prices: List[int]) -> int:
    min_price = float("inf")
    max_profit = 0
    for p in prices:
        if p < min_price:
            min_price = p
        else:
            max_profit = max(max_profit, p - min_price)
    return max_profit


def main() -> None:
    import sys

    data = sys.stdin.read().strip().split()
    prices = list(map(int, data)) if data else []
    print(max_profit(prices))


if __name__ == "__main__":
    main()

